### PR TITLE
[ticket/13311] Add core event on acp_forums for forum content deletes

### PR DESCRIPTION
**Pull request by Alf007**
Event added for the function delete_forum_content() in
includes/acp/acp_forums.php, so that you can insert any extension's table(s)
to be updated when forum content is deleted

PHPBB3-13311

### CODE REVIEW

File: phpBB/includes/acp/acp_forums.php [hunk 1]
```diff
@@ -1918,6 +1918,18 @@ function delete_forum_content($forum_id)
...
+		/**
+		 * Perform additional actions before forum content deletion
+		 *
+		 * @event core.delete_forum_content_before_query
+		 * @var	array	table_ary	Array of tables from which all rows will be deleted that hold the forum_id
+		 * @since 3.1.5-RC1
+		 */
+		$vars = array(
+				'table_ary',
+		);
```
Comment (marc1706) [inline phpBB/includes/acp/acp_forums.php]: You should add more variables to this event, in this case I'd say $forum_id, $topic_ids, and $post_counts might also be helpful for other extension authors.

Comment (Alf007) [inline phpBB/includes/acp/acp_forums.php]: In this function, only $forum_id is available, and it won't be of any real use for an extension in this context.

Comment (nickvergessen) [inline phpBB/includes/acp/acp_forums.php]: Please add them. I know extensions that might be interested into those

**Comment by Alf007** [comment]
Incuded parameters forum_id, topic_ids and post_counts

**Comment by Alf007** [comment]
Had forgotten real changes ^^

File: phpBB/includes/acp/acp_forums.php [hunk 2]
```diff
@@ -1918,6 +1918,24 @@ function delete_forum_content($forum_id)
...
+		 * @var	array	table_ary	Array of tables from which all rows will be deleted that hold the forum_id
+		 * @var	int		forum_id	the forum id
+		 * @var	array	topic_ids	Array of the topic ids from the forum to be deleted 	
+		 * @var	array	post_counts	Array of counts of posts in the forum, by poster_id
+		 * @since 3.1.5-RC1
+		 */
+		$vars = array(
+				'table_ary',
+				'forum_id',
+				'topic_ids'
```
Comment (nickvergessen) [inline phpBB/includes/acp/acp_forums.php]:
```
There was 1 error:
1) phpbb_event_export_php_test::test_crawl_php_file with data set #398 ('includes/acp/acp_forums.php')

LogicException: Found invalid var 'topic_ids, 'forum_id' in array for event 'core.delete_forum_content_before_query' in file 'includes/acp/acp_forums.php:1936'

/home/travis/build/phpbb/phpbb/phpBB/phpbb/event/php_exporter.php:316
/home/travis/build/phpbb/phpbb/phpBB/phpbb/event/php_exporter.php:194
/home/travis/build/phpbb/phpbb/tests/event/export_php_test.php:47
```

Missing `,`

**Comment by marc1706** [comment]
Could you please squash your commits? If you don't know how to do this, just let me know and I'll do this for you.

File: phpBB/includes/acp/acp_forums.php [hunk 3]
```diff
@@ -1918,6 +1918,24 @@ function delete_forum_content($forum_id)
...
 
+		/**
+		 * Perform additional actions before forum content deletion
+		 *
+		 * @event core.delete_forum_content_before_query
+		 * @var	array	table_ary	Array of tables from which all rows will be deleted that hold the forum_id
+		 * @var	int		forum_id	the forum id
+		 * @var	array	topic_ids	Array of the topic ids from the forum to be deleted 	
+		 * @var	array	post_counts	Array of counts of posts in the forum, by poster_id
+		 * @since 3.1.5-RC1
```
Comment (nickvergessen) [inline phpBB/includes/acp/acp_forums.php]: 3.1.6-RC1 btw

**Comment by Alf007** [comment]
Is this ok?

**Comment by nickvergessen** [comment]
@marc1706 just solve the remaining issues on merge? ;)

**Comment by marc1706** [comment]
Closing in favor of #3786

**Pull request closed without merging**